### PR TITLE
Dockerfile converted into multistage

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
**Description:**

Added two stages dev and prod target in Dockerfile.

**Steps to test**
- Clone the repo
- Checkout to zshkoor/multi-stage-dockerfile branch
- Use this command to createe an IDA
`cookiecutter -o /path/to/folder/for/creating/project cookiecutter-django-ida`
- Now go to project's root and run `make upgrade` to prepare requirements files
- Now run `make docker_build` to make test image build

**Issue:**

https://github.com/openedx/edx-cookiecutters/issues/292

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [x] Commits are squashed

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]
These changes looks good to go.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile [hunk 1]
```diff
@@ -42,29 +42,49 @@ RUN locale-gen en_US.UTF-8
...
 # this prevents the image cache from busting unless the dependencies have changed.
-COPY requirements/production.txt /edx/app/{{cookiecutter.repo_name}}/requirements/production.txt
+COPY requirements/development.txt /edx/app/{{cookiecutter.repo_name}}/requirements/development.txt
 
 # Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip install -r requirements/production.txt
+RUN pip install -r requirements/development.txt
 
-RUN mkdir -p /edx/var/log
+# Gunicorn 19 does not log to stdout or stderr by default. Once we are past gunicorn 19, the logging to STDOUT need not be specified.
```
Comment (iamsobanjaved) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:67]: This comment isn't relevant here.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile [hunk 2]
```diff
@@ -42,29 +42,49 @@ RUN locale-gen en_US.UTF-8
...
+# Dependencies are installed as root so they cannot be modified by the application user.
+RUN pip install -r requirements/production.txt
 
 # Gunicorn 19 does not log to stdout or stderr by default. Once we are past gunicorn 19, the logging to STDOUT need not be specified.
 CMD gunicorn --workers=2 --name {{cookiecutter.repo_name}} -c /edx/app/{{cookiecutter.repo_name}}/{{cookiecutter.project_name}}/docker_gunicorn_configuration.py --log-file - --max-requests=1000 {{cookiecutter.project_name}}.wsgi:application
 
 # This line is after the requirements so that changes to the code will not
 # bust the image cache
-COPY . /edx/app/{{cookiecutter.repo_name}}
+COPY . /edx/app/{{cookiecutter.repo_name}}
```
Comment (iamsobanjaved) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:90]: Add an empty line at the end of the file.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile [hunk 3]
```diff
@@ -42,29 +42,49 @@ RUN locale-gen en_US.UTF-8
...
-COPY requirements/production.txt /edx/app/{{cookiecutter.repo_name}}/requirements/production.txt
+COPY requirements/development.txt /edx/app/{{cookiecutter.repo_name}}/requirements/development.txt
 
 # Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip install -r requirements/production.txt
+RUN pip install -r requirements/development.txt
 
-RUN mkdir -p /edx/var/log
+# Gunicorn 19 does not log to stdout or stderr by default. Once we are past gunicorn 19, the logging to STDOUT need not be specified.
+CMD while true; do python ./manage.py runserver 0.0.0.0:8120; sleep 2; done
```
Comment (iamsobanjaved) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:68]: Instead of hardcoding the port, use cookie-cutter input `{{cookiecutter.port}}`

**Review by iamsobanjaved — COMMENTED** [review]
While enhancing native Dockerfiles to use to build docker images for devstack, we have set some standards which we want to replicate in cookiecutter here. From those standards was the use of  `virtualenv` to install python requirements. We also need to set up `virtualenv` here so we have consistency among IDAs.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile (hunk 1 above)
Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile]: removed

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile (hunk 2 above)
Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile]: added

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile (hunk 3 above)
Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile]: done

Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile]: > While enhancing native Dockerfiles to use to build docker images for devstack, we have set some standards which we want to replicate in cookiecutter here. From those standards was the use of `virtualenv` to install python requirements. We also need to set up `virtualenv` here so we have consistency among IDAs.

Thanks @iamsobanjaved for the suggestion. I've added virtual environment and tested its successful docker image build on local.

**Comment by zubairshakoorarbisoft** [comment]
Closing this PR as an other [PR](https://github.com/openedx/edx-cookiecutters/pull/395/files) has already been merged which is doing the same.

**Pull request closed without merging**